### PR TITLE
Add lint to validate tgstation.dme is in the proper order

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds `validate_dme.py` to the CI suite to verify that tgstation.dme is in proper alphabetical order

And by alphabetical order I mean the ass-backwards specific rules that Dream Maker enforces

### CODE REVIEW

File: tools/validate_dme.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        if a_segment != b_segment:
+            return (a_segment > b_segment) - (a_segment < b_segment)
+
+    raise f"Two lines were exactly the same ({a} vs. {b})"
+
+sorted_lines = sorted(lines, key = functools.cmp_to_key(compare_lines))
+
+for (index, line) in enumerate(lines):
+    if sorted_lines[index] != line:
+        print(f"The include at line {index + 1} is out of order ({line})")
```
Comment (Tastyfish) [inline tools/validate_dme.py:50]: If you also print the following to the console, it should also create a github annotation on the offending line
```python
print(f"::error file=tgstation.dme,line={index+1},title=DME Validator::The include at line {index + 1} is out of order ({line})")
```

Comment (Tastyfish) [reply tools/validate_dme.py:50]: If you want to do that, that is

Comment (Mothblocks) [reply tools/validate_dme.py:50]: Alright, make it a code suggestion and I'll merge it

Comment (Tastyfish) [reply tools/validate_dme.py:50]:
```suggestion
        print(f"The include at line {index + 1} is out of order ({line})")
        print(f"::error file=tgstation.dme,line={index+1},title=DME Validator::The include at line {index + 1} is out of order ({line})")
```

Comment (Mothblocks) [reply tools/validate_dme.py:50]: Thanks!

**Pull request merged**